File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, bool> mp;
        for (auto num: nums) {
            if (mp[num]) return true;
            mp[num] = true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        final Map<Integer, Integer> countMap = new HashMap<>();
        for (final int num : nums) {
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }
        final List<Integer> result = new ArrayList<>();
        countMap.forEach((key, value) -> {
            if (value >= k) {
                result.add(key);
            }
        });
        return result.stream().mapToInt(i -> i).toArray();  
    }
}
